internal/parser.go

```diff
@@ -2763,7 +2763,7 @@ func (p *parser) parseCurrentToken() {
 		if p.inForeignContent() {
 			consumed = parseForeignContent(p)
 		} else {
-			if p.im == nil && p.context != nil {
+			if p.im == nil {
```

Comment:
This is the fix! Special elements like `iframe` and `noscript` were setting `p.context` which meant `p.im(p)` below would throw.